### PR TITLE
cephadm: support no huge pages for nvmeof spdk

### PR DESCRIPTION
**Pull request by baum**
# cephadm: do not use huge pages for nvmeof spdk

- depends on: https://github.com/ceph/ceph-nvmeof/pull/898
- shaman build: https://shaman.ceph.com/builds/ceph/wip-baum-no-huge-20241120-02/acf20409eb236f8fe9c237dc5b64a4e2ed7811cf/
- gateway CI integration: https://github.com/ceph/ceph-nvmeof/pull/959
   - no-huge cephadm sanity test: https://github.com/ceph/ceph-nvmeof/actions/runs/11931868784/job/33258685579?pr=959

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/python-common/ceph/deployment/service_spec.py [hunk 1]
```diff
@@ -1345,7 +1345,7 @@ def __init__(self,
                  client_key: Optional[str] = None,
                  client_cert: Optional[str] = None,
                  root_ca_cert: Optional[str] = None,
-                 spdk_path: Optional[str] = None,
```
Comment (adk3798) [inline src/python-common/ceph/deployment/service_spec.py:1348]: I think we previously had a discussion about removing fields from the spec file class and the risks of that. If someone had `spdk_path` set in their spec file and then upgraded to a version that removed it, cephadm will fail to load the spec causing the service and eventually the daemons to be removed. Do we know if this field was present in any released versions of ceph?

File: src/cephadm/cephadmlib/daemons/nvmeof.py [hunk 2]
```diff
@@ -222,4 +214,4 @@ def customize_container_args(
         args.append(ctx.container_engine.unlimited_pids_option)
         args.extend(['--ulimit', 'memlock=-1:-1'])
         args.extend(['--ulimit', 'nofile=10240'])
-        args.extend(['--cap-add=SYS_ADMIN', '--cap-add=CAP_SYS_NICE'])
+        args.extend(['--cap-add=CAP_SYS_NICE'])
```
Comment (adk3798) [inline src/cephadm/cephadmlib/daemons/nvmeof.py:217]: you'll have to update the unit tests this affects as well
```
FAILED tests/test_daemon_form.py::test_is_sysctl_daemon_form[nvmeof-True] - A...
FAILED tests/test_deploy.py::test_deploy_nvmeof_container - AssertionError: a...
```

**Comment by baum** [comment]
jenkins retest this please

File: src/python-common/ceph/deployment/service_spec.py [hunk 3]
```diff
@@ -1347,7 +1347,8 @@ def __init__(self,
...
+                 spdk_path: Optional[str] = None, # unused and duplicate of tgt_path bellow, consider removing
```
Comment (adk3798) [inline src/python-common/ceph/deployment/service_spec.py:1350]:
```suggestion
                 spdk_path: Optional[str] = None, # unused and duplicate of tgt_path below, consider removing
```
NIT, typo

**Review by adk3798 — APPROVED** [review]
patch seems good to me

**Comment by adk3798** [comment]
jenkins test make check

**Comment by baum** [comment]
jenkins test make check
jenkins test make check arm64
jenkins test dashboard

**Comment by baum** [comment]
jenkins test make check

**Comment by baum** [comment]
jenkins test dashboard

**Comment by baum** [comment]
✅ ceph-nvmeof atom e2e test - https://github.com/ceph/ceph-nvmeof/actions/runs/11931868784/job/33258682992?pr=959

**Comment by baum** [comment]
jenkins test make check

**Comment by baum** [comment]
jenkins test dashboard

**Comment by baum** [comment]
jenkins test dashboard cephadm

**Comment by adk3798** [comment]
@VallariAg I think this is ready for another run through the nvmeof teuthology suite

**Comment by baum** [comment]
jenkins test make check arm64

**Comment by VallariAg** [comment]
nvmeof suite passed! 
with default config:
https://pulpito.ceph.com/vallariag-2024-11-20_11:21:39-nvmeof-wip-baum-no-huge-20241120-02-distro-default-smithi/
https://pulpito.ceph.com/vallariag-2024-11-20_14:33:42-nvmeof-wip-baum-no-huge-20241120-02-distro-default-smithi/
with no huge pages config: https://pulpito.ceph.com/vallariag-2024-11-21_07:48:13-nvmeof-wip-baum-no-huge-20241120-02-distro-default-smithi/

**Review by caroav — APPROVED** [review]

**Pull request merged**